syntax = "proto2";

package NCloud.NProto;

option go_package = "github.com/ydb-platform/nbs/cloud/storage/core/config";

////////////////////////////////////////////////////////////////////////////////

message TFilters
{
    repeated string CloudIds = 1;
    repeated string FolderIds = 2;

    // DiskId for blockstore or FsId for filestore.
    repeated string EntityIds = 3;
}

////////////////////////////////////////////////////////////////////////////////

message TFeatureConfig
{
    // Feature name.
    optional string Name = 1;

    // Enable feature for Clouds and Folders from this list.
    optional TFilters Whitelist = 2;

    // Disable feature for Clouds and Folders from this list.
    optional TFilters Blacklist = 3;

    // Optional value - for non-binary features.
    optional string Value = 4;

    // By setting these probabilities, feature can be enabled for Ids that
    // aren't in the whitelist. If whitelist is not specified, default value
    // is 1 for backwards compatibility, otherwise 0.
    optional double CloudProbability = 5;
    optional double FolderProbability = 6;
};

////////////////////////////////////////////////////////////////////////////////

message TFeaturesConfig
{
    repeated TFeatureConfig Features = 1;
}
